### PR TITLE
Added test for route, created a merchant items controller

### PR DESCRIPTION
**Pull request by MDelarosa1993**
Add MerchantItemsController to Handle Merchant Retrieval for Items

- Implemented MerchantItemsController to fetch and return the merchant associated with a specific item.
- Updated routes to map /api/v1/items/:id/merchant to the new controller action.
- Added request spec to test the new endpoint.
- Modified existing test files for consistency.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -14,13 +14,10 @@
...
-  
-  
   get "/api/v1/items", to: "api/v1/items#index"
   get "/api/v1/items/:id", to: "api/v1/items#show"
   patch "/api/v1/items/:id", to: "api/v1/items#update" 
-
   post "/api/v1/items", to: "api/v1/items#create"
-  get "/api/v1/items?sorted=price", to: "api/v1/items#index"
 
+  get '/api/v1/items/:id/merchant', to: 'api/v1/merchant_items#show'
```
Comment (CCirbo) [inline config/routes.rb:22]: endpoint looks great! Did you test with Postman?

File: spec/requests/api/v1/merchants_requests_spec.rb [hunk 2]
```diff
@@ -22,7 +22,7 @@
     merchants = JSON.parse(response.body, symbolize_names: true)[:data]
     
     expect(response).to be_successful
-    expect(merchants.count).to eq(0) #had to change this from 3 to 0 to get test to pass
+    expect(merchants.count).to eq(3) #had to change this from 3 to 0 to get test to pass
```
Comment (CCirbo) [inline spec/requests/api/v1/merchants_requests_spec.rb:25]: Thank you for updating here and the other place. Looks great!

File: spec/requests/merchant_items_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
```
Comment (CCirbo) [inline spec/requests/merchant_items_request_spec.rb:2]: Good coverage on the testing!

**Comment by CCirbo** [comment]
I added individual comments, great job!

File: config/routes.rb (hunk 1 above)
Comment (MDelarosa1993) [reply config/routes.rb:22]: Seems like it, ran it a couple a times and i dont get any errors

Comment (MDelarosa1993) [reply config/routes.rb:22]: Double checked on Postico as well

**Review by TDManning — APPROVED** [review]
Thanks for the updates, Melchor.

**Comment by TDManning** [comment]
Reviewed and confirmed by two team mates.

**Pull request merged**